<a href="https://colab.research.google.com/github/RogerMoi26/Discrete_Math-1/blob/master/RogerMoison_CPSMA3913_ProjectPart_4_Counting.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project Part # 4: CPSMA 3133 [Dr. Jacob's](https://github.com/nurfnick) Discrete Mathematics 

## Counting

### Roger Moison and 282541




Okay this week is a fun one... so lets jump in shall we.
last week i used my daughter as the example but i dont think that will work for this one. but imma do it anyway..

There are *7* possible gifts my daughter will recieve for her birthday and *10* presents to chose from

In [52]:
possibleChoices = ['Shirts','Stuffed Animal','Shoes','Money','Books','Pants','Socks']

## 1 (a) 

$$ 7*7*7\dots *7 = 7^{10} $$

Now i did not wrap the gifts so i dont know what is in which gift.

In [53]:
7**10

282475249

#(b) 




$$
\binom{n+k-1}{k-1} = \binom{n+k-1}{n}
$$



In [54]:
import math

math.factorial(16)/(math.factorial(10)*math.factorial(6))

8008.0

## (c) 

In [55]:
from itertools import combinations_with_replacement

print(list(combinations_with_replacement(possibleChoices,10)))

[('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Stuffed Animal'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shoes'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Money'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Books'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Pants'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Socks'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Stuffed Animal', 'Stuffed Animal'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Stuffed Animal', 'Shoes'), ('Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shirts', 'Shi

In [56]:
len(list(combinations_with_replacement(possibleChoices,10)))

8008

In [57]:
print(math.factorial(7))

5040


and out of those 7 gift choices I have 5040 possibilities.

When order matters

In [60]:
from itertools import permutations
print(list(permutations(possibleChoices,7)))

[('Shirts', 'Stuffed Animal', 'Shoes', 'Money', 'Books', 'Pants', 'Socks'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Money', 'Books', 'Socks', 'Pants'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Money', 'Pants', 'Books', 'Socks'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Money', 'Pants', 'Socks', 'Books'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Money', 'Socks', 'Books', 'Pants'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Money', 'Socks', 'Pants', 'Books'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Books', 'Money', 'Pants', 'Socks'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Books', 'Money', 'Socks', 'Pants'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Books', 'Pants', 'Money', 'Socks'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Books', 'Pants', 'Socks', 'Money'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Books', 'Socks', 'Money', 'Pants'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Books', 'Socks', 'Pants', 'Money'), ('Shirts', 'Stuffed Animal', 'Shoes', 'Pants', 'Money', 'Books', 'Socks'), ('Shirts', 'Stuffed Anim

When it does not

In [61]:
math.factorial(7)/(math.factorial(0)*math.factorial(7))

1.0

In [62]:
from itertools import combinations
print(list(combinations(possibleChoices,7)))

[('Shirts', 'Stuffed Animal', 'Shoes', 'Money', 'Books', 'Pants', 'Socks')]


# (d)

What is the coefficent of the 5th order of (x-3)^9?
So lets take 9C5

I hated this so the answer would be 126